build/charts/theia/provisioning/datasources/migrate.sh

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 Antrea Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function checkDataVersion {
+    tables=$(clickhouse client -h 127.0.0.1 -q "SHOW TABLES")
+    if [[ $tables == *"migrate_version"* ]]; then
+        dataVersion=$(clickhouse client -h 127.0.0.1 -q "SELECT version FROM migrate_version")
+    elif [[ $tables == *"flows"* ]]; then
```

Comment:
Do we need to explicitly check for *"flows"*?
(this is not important, I'm just trying to correctly understand the logic)

Reply:
I think we need to check this to confirm whether there are data schema created before. When we find there is no `migrate_version` table, there might be two conditions:

- We are upgrading from v0.1. In this case, we can find the table `flows` in database. And we need to run the upgrading SQLs.
- We are deploying ClickHouse with an empty PersistentVolume. There is no data schema created before. In this case, we cannot find the table `flows`. And we can stop the migration and create the data schema.

Reply:
Understood, so we could be checking for 'flows' as well as for _any_ table in the DB.
Makes sense.